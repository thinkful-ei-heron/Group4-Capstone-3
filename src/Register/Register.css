
body {
    background-image: url("../images/beer3.jpg");
    background-repeat: no-repeat;
    padding: 90px 200px;
  
}


h1 {
    font-family: 'Merriweather', serif;
    color: rgb(204, 192, 172);
    font-size: 45px;
   
}

h3 {
    color: white;
}

div {
    background-color: rgba(0, 0, 0, 0.365);

}

label {
    color: whitesmoke;
}