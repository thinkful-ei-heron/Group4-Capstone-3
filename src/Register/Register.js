import React from 'react';
import AuthApiService from "../services/auth-api-service";
import './Register.css' 
import TokenService from '../services/token-service';

class Register extends React.Component {
    state = { error: null };

    handleSubmit = ev => {
        ev.preventDefault();
        const {nickname, username, password, confirmPassword} = ev.target;

        this.setState({error: null});
        if(password.value === confirmPassword.value) {
            AuthApiService.postUser({
                user_name: username.value,
                password: password.value,
                full_name: nickname.value,
            })
                .then(() => {
                    return AuthApiService.postLogin({
                        user_name: username.value,
                        password: password.value,
                    })
                })
                .then(res => {
                    TokenService.saveAuthToken(res.authToken)
                    window.location.replace('/home')
                    username.value = '';
                    password.value = '';
                    confirmPassword.value = '';
                    nickname.value = '';
                })
                .catch(res => {
                    this.setState({error: res.error})
                })
        } else {
            this.setState({error: 'Passwords dont Match'});
        }
    };

    render() {
        const {error} = this.state;
        return (
        <div className="register_route">
            <div className="register">
                <h1>DearBeer</h1>
                <h3>Register</h3>

                <form
                    className='RegistrationForm'
                    onSubmit={this.handleSubmit}
                >
                    <div role='alert'>
                        {error && <p className='error'>{error}</p>}
                    </div>
                    <div className='username'>
                        <label htmlFor='RegistrationForm__username'>
                            {/* Username */}
                        </label>
                        <input
                            name='username'
                            type='text'
                            required
                            id='RegistrationForm__username'
                            placeholder= 'Username'>
                        </input>
                    </div>
                    <div className='password'>
                        <label htmlFor='RegistrationForm__password'>
                            {/* Password */}
                        </label>
                        <input
                            name='password'
                            type='password'
                            required
                            id='RegistrationForm__password'
                            placeholder= 'Password'>
                        </input>
                    </div>
                    <div className='password'>
                        <label htmlFor='RegistrationForm__confirmPassword'>
                            {/* Confirm Password */}
                        </label>
                        <input
                            name='confirmPassword'
                            type='password'
                            required
                            id='RegistrationForm__confirmPassword'
                            placeholder= 'Confirm Password'>
                        </input>
                    </div>
                    <div className='full_name'>
                        <label htmlFor='RegistrationForm__nickname'>
                            {/* Nickname */}
                        </label>
                        <input
                            name='nickname'
                            type='text'
                            required
                            id='RegistrationForm__nickname'
                            placeholder= 'Nick Name'>
                        </input>
                    </div>
                    <button className='submit-button' type='submit'>
                        Submit
                    </button>
                    <button className='home-btn' type='button' onClick={()=>window.location.replace('/login')}>
                        Home
                    </button>
                </form>
            </div>
            </div>
        )
    }
}

export default Register;