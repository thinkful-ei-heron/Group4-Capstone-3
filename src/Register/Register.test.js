import React from 'react';
import ReactDOM from 'react-dom';
import Register from './Register';
import { BrowserRouter } from 'react-router-dom';

describe('Register Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render
      (<BrowserRouter> 
        <Register /> 
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  
});
