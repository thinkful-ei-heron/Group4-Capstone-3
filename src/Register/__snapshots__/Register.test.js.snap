// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Component Displays Register when rendered 1`] = `
<div
  className="register_route"
>
  <div
    className="register"
  >
    <Link
      className="register-header"
      to="/"
    >
      <h1>
        DearBeer
      </h1>
    </Link>
    <h3>
      Register
    </h3>
    <form
      className="RegistrationForm"
      onSubmit={[Function]}
    >
      <div
        className="alert"
        role="alert"
      />
      <div
        className="username"
      >
        <label
          htmlFor="RegistrationForm__username"
        />
        <input
          id="RegistrationForm__username"
          name="username"
          placeholder="Username"
          required={true}
          type="text"
        />
      </div>
      <div
        className="password"
      >
        <label
          htmlFor="RegistrationForm__password"
        />
        <input
          id="RegistrationForm__password"
          name="password"
          placeholder="Password"
          required={true}
          type="password"
        />
      </div>
      <div
        className="password"
      >
        <label
          htmlFor="RegistrationForm__confirmPassword"
        />
        <input
          id="RegistrationForm__confirmPassword"
          name="confirmPassword"
          placeholder="Confirm Password"
          required={true}
          type="password"
        />
      </div>
      <div
        className="full_name"
      >
        <label
          htmlFor="RegistrationForm__nickname"
        />
        <input
          id="RegistrationForm__nickname"
          name="nickname"
          placeholder="Nick Name"
          required={true}
          type="text"
        />
      </div>
      <button
        className="submit-button"
        type="submit"
      >
        Submit
      </button>
      <br />
      <button
        className="login-btn"
        onClick={[Function]}
        type="button"
      >
        Already a user?
      </button>
    </form>
  </div>
</div>
`;
